src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_targetid.lua

```diff
@@ -106,9 +106,14 @@ function GM:DrawSigilTargetHint(ent, fade)
 	util.ColorCopy(color_white, colTemp)
 
 	draw.SimpleTextBlur("Sigil", "ZSHUDFontSmaller", x, y, colTemp, TEXT_ALIGN_CENTER)
-	y = y + draw.GetFontHeight("ZSHUDFontSmaller") + 0
+	local fontHeight = draw.GetFontHeight("ZSHUDFontSmaller")
+	y = y + fontHeight + 0
 
 	draw.SimpleTextBlur("Press E to teleport", "ZSHUDFontTiny", x, y, colTemp, TEXT_ALIGN_CENTER)
+	
+	if self.GetWave() > 0 then
```

Comment:
I would consider applying a check that uses the `GM:PlayerCanPurchase` hook.

https://github.com/rbreslow/zs/blob/00f1a80f382d72ceb91912941587e67b5b735e05/src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_init.lua#L911-L913